core/controllers/profile.py

```diff
@@ -354,15 +354,21 @@ def put(self) -> None:
                     'can_receive_editor_role_email',
                     'can_receive_feedback_message_email',
                     'can_receive_subscription_email']
-                # Check if all required keys are in the data dictionary.
-                if not all(key in data for key in required_keys):
+                missing_keys = [
+                    key for key in required_keys if key not in data]
+                if missing_keys:
                     raise self.InvalidInputException(
-                        'Expected data to contain the fields,%s, received %s'
-                        % (required_keys, data))
-                if not all(isinstance(value, bool) for value in data.values()):
+                        'Expected data dict to have all the required keys.'
```

Comment:
Add a space at the end of the string, otherwise you will get "keys.Missing" in the string.

Reply:
Done.